src/darwin/Framework/CHIP/MTROTAUnsolicitedBDXMessageHandler.mm

```diff
@@ -0,0 +1,129 @@
+    mExchangeMgr->UnregisterUnsolicitedMessageHandlerForProtocol(Protocols::BDX::Id);
+
+    VerifyOrReturn(mOTAImageTransferHandler != nullptr);
+
+    delete mOTAImageTransferHandler;
+    mOTAImageTransferHandler = nullptr;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::ControllerShuttingDown(MTRDeviceController_Concrete * controller)
+{
+    assertChipStackLockedByCurrentThread();
+
+    VerifyOrReturn(mOTAImageTransferHandler != nullptr);
+
+    FabricIndex peerFabricIndex = mOTAImageTransferHandler->GetPeerFabricIndex();
+    VerifyOrReturn(peerFabricIndex != kUndefinedFabricIndex);
+
+    if (controller.fabricIndex == peerFabricIndex)
+    {
+        delete mOTAImageTransferHandler;
+        mOTAImageTransferHandler = nullptr;
+    }
+}
+
+bool MTROTAUnsolicitedBDXMessageHandler::IsInAnOngoingTransfer()
+{
+    return (mOTAImageTransferHandler != nullptr);
+}
+
+CHIP_ERROR MTROTAUnsolicitedBDXMessageHandler::OnUnsolicitedMessageReceived(const PayloadHeader & payloadHeader, const SessionHandle & session,
+    ExchangeDelegate * _Nonnull & newDelegate)
+{
+    assertChipStackLockedByCurrentThread();
+
+    ChipLogDetail(BDX, "MTROTAUnsolicitedBDXMessageHandler: OnUnsolicitedMessageReceived: message " ChipLogFormatMessageType " protocol " ChipLogFormatProtocolId,
+        payloadHeader.GetMessageType(), ChipLogValueProtocolId(payloadHeader.GetProtocolID()));
+
+    VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    // Only proceed if there is a valid fabric index for the SessionHandle.
+    if (session->IsSecureSession() && session->AsSecureSession() != nullptr && session->AsSecureSession()->GetFabricIndex() != kUndefinedFabricIndex) {
+
+        // If we receive a ReceiveInit BDX message, create a new MTROTAImageTransferHandler and register it
+        // as the handler for all BDX messages that will come over this exchange.
+        if (payloadHeader.HasMessageType(MessageType::ReceiveInit)) {
+            mOTAImageTransferHandler = new MTROTAImageTransferHandler(mSystemLayer);
+            newDelegate = mOTAImageTransferHandler;
+            return CHIP_NO_ERROR;
+        }
+    }
+    return CHIP_ERROR_INCORRECT_STATE;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::OnExchangeCreationFailed(ExchangeDelegate * delegate)
+{
+    assertChipStackLockedByCurrentThread();
+    auto * otaTransferHandler = static_cast<MTROTAImageTransferHandler *>(delegate);
+    VerifyOrReturn(otaTransferHandler != nullptr);
+
+    delete otaTransferHandler;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::OnTransferHandlerCreated(void * imageTransferHandler)
+{
+    assertChipStackLockedByCurrentThread();
+
+    // TODO: #36181 - Store the imageTransferHandler in a set of MTROTAImageTransferHandler objects.
+    mOTAImageTransferHandler = static_cast<MTROTAImageTransferHandler *>(imageTransferHandler);
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::OnTransferHandlerDestroyed(void * imageTransferHandler)
```

Comment:
Again, this arg should just be `MTROTAImageTransferHandler *` if that's what we need.

Reply:
done.